No matching export in "../../node_modules/@mui/x-data-grid-pro/esm/index.js" for import "GridEventLookup"

### Steps to reproduce

Steps:
1. import x-data-grid-premium


### Current behavior

using x-data-grid-premium is importing a component from x-data-grid-pro, causing an error in versione 8.26.0

X [ERROR] No matching export in "../../node_modules/@mui/x-data-grid-pro/esm/index.js" for import "GridEventLookup"
    ../../node_modules/@mui/x-data-grid-premium/esm/typeOverloads/modules.js:1:9:
      1 │ import { GridEventLookup } from '@mui/x-data-grid-pro';

### Expected behavior

GridEventLookup should be imported from premium grid

### Context

_No response_

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  System:
    OS: macOS 26.2
  Binaries:
    Node: 22.20.0 - /Users/alessiobrunello/.nvm/versions/node/v22.20.0/bin/node
    npm: 10.9.3 - /Users/alessiobrunello/.nvm/versions/node/v22.20.0/bin/npm
    pnpm: 10.25.0 - /opt/homebrew/bin/pnpm
  Browsers:
    Chrome: 144.0.7559.59
    Edge: Not Found
    Firefox: 145.0.2
    Safari: 26.2
  npmPackages:
    @base-ui/utils:  0.2.4 
    @emotion/react: ^11.10.5 => 11.14.0 
    @emotion/styled: ^11.10.5 => 11.14.1 
    @mui/core-downloads-tracker:  7.3.7 
    @mui/icons-material: ^7.1.0 => 7.3.7 
    @mui/material: ^7.1.0 => 7.3.7 
    @mui/private-theming:  7.3.7 
    @mui/styled-engine:  7.3.7 
    @mui/system:  7.3.7 
    @mui/types:  7.4.10 
    @mui/utils:  7.3.7 
    @mui/x-data-grid:  8.25.0 
    @mui/x-data-grid-premium: 8.25.0 => 8.25.0 
    @mui/x-data-grid-pro:  8.25.0 
    @mui/x-date-pickers:  8.25.0 
    @mui/x-date-pickers-pro: 8.25.0 => 8.25.0 
    @mui/x-internal-exceljs-fork:  4.4.3 
    @mui/x-internals:  8.25.0 
    @mui/x-license:  8.25.0 
    @mui/x-telemetry:  8.20.0 
    @mui/x-tree-view: 8.25.0 => 8.25.0 
    @mui/x-virtualizer:  0.3.2 
    @types/react: 18.3.1 => 18.3.1 
    react: 18.3.1 => 18.3.1 
    react-dom: 18.3.1 => 18.3.1 
    styled-components:  6.3.8 
    typescript: ^5.9.3 => 5.9.3 
```
</details>



**Search keywords**: No matching export in "../../node_modules/@mui/x-data-grid-pro/esm/index.js" for import "GridEventLookup"

--- a/packages/x-data-grid-premium/src/typeOverloads/modules.ts
+++ b/packages/x-data-grid-premium/src/typeOverloads/modules.ts
@@ -172,19 +172,19 @@ export interface GridPipeProcessingLookupPremium {
 }
 
 declare module '@mui/x-data-grid-pro' {
-  interface GridEventLookup extends GridEventLookupPremium { }
+  interface GridEventLookup extends GridEventLookupPremium {}
 
   interface GridPipeProcessingLookup
-    extends GridPipeProcessingLookupPro, GridPipeProcessingLookupPremium { }
+    extends GridPipeProcessingLookupPro, GridPipeProcessingLookupPremium {}
 
   interface GridControlledStateEventLookup
-    extends GridControlledStateEventLookupPro, GridControlledStateEventLookupPremium { }
+    extends GridControlledStateEventLookupPro, GridControlledStateEventLookupPremium {}
 
-  interface GridRenderCellParams<R, V, F> extends GridRenderCellParamsPremium<R, V, F> { }
+  interface GridRenderCellParams<R, V, F> extends GridRenderCellParamsPremium<R, V, F> {}
 
-  interface GridColumnHeaderParams<R, V, F> extends GridColumnHeaderParamsPremium<R, V, F> { }
+  interface GridColumnHeaderParams<R, V, F> extends GridColumnHeaderParamsPremium<R, V, F> {}
 
-  interface GridApiCaches extends GridApiCachesPremium { }
+  interface GridApiCaches extends GridApiCachesPremium {}
 
   interface GridToolbarExportProps {
     excelOptions?: GridExcelExportOptions & GridExportDisplayOptions;
@@ -196,7 +196,7 @@ declare module '@mui/x-data-grid-pro' {
 }
 
 declare module '@mui/x-data-grid-pro/internals' {
-  interface GridApiCaches extends GridApiCachesPremium { }
+  interface GridApiCaches extends GridApiCachesPremium {}
 
-  interface GridBaseColDef<R, V, F> extends GridColDefPremium<R, V, F> { }
+  interface GridBaseColDef<R, V, F> extends GridColDefPremium<R, V, F> {}
 }

--- a/packages/x-data-grid-premium/src/typeOverloads/modules.ts
+++ b/packages/x-data-grid-premium/src/typeOverloads/modules.ts
@@ -1,4 +1,4 @@
-import {
+import type {
   GridEventLookup,
   GridExportDisplayOptions,
   GridRowId,
@@ -172,19 +172,19 @@ export interface GridPipeProcessingLookupPremium {
 }
 
 declare module '@mui/x-data-grid-pro' {
-  interface GridEventLookup extends GridEventLookupPremium {}
+  interface GridEventLookup extends GridEventLookupPremium { }
 
   interface GridPipeProcessingLookup
-    extends GridPipeProcessingLookupPro, GridPipeProcessingLookupPremium {}
+    extends GridPipeProcessingLookupPro, GridPipeProcessingLookupPremium { }
 
   interface GridControlledStateEventLookup
-    extends GridControlledStateEventLookupPro, GridControlledStateEventLookupPremium {}
+    extends GridControlledStateEventLookupPro, GridControlledStateEventLookupPremium { }
 
-  interface GridRenderCellParams<R, V, F> extends GridRenderCellParamsPremium<R, V, F> {}
+  interface GridRenderCellParams<R, V, F> extends GridRenderCellParamsPremium<R, V, F> { }
 
-  interface GridColumnHeaderParams<R, V, F> extends GridColumnHeaderParamsPremium<R, V, F> {}
+  interface GridColumnHeaderParams<R, V, F> extends GridColumnHeaderParamsPremium<R, V, F> { }
 
-  interface GridApiCaches extends GridApiCachesPremium {}
+  interface GridApiCaches extends GridApiCachesPremium { }
 
   interface GridToolbarExportProps {
     excelOptions?: GridExcelExportOptions & GridExportDisplayOptions;
@@ -196,7 +196,7 @@ declare module '@mui/x-data-grid-pro' {
 }
 
 declare module '@mui/x-data-grid-pro/internals' {
-  interface GridApiCaches extends GridApiCachesPremium {}
+  interface GridApiCaches extends GridApiCachesPremium { }
 
-  interface GridBaseColDef<R, V, F> extends GridColDefPremium<R, V, F> {}
+  interface GridBaseColDef<R, V, F> extends GridColDefPremium<R, V, F> { }
 }

